### DISCUSSION TITLE
有什么办法从 Antd 4 平滑过渡到 5

### DISCUSSION THREAD

**Discussion by zhangminggeek** [ant-design/ant-design#40720] [Q&A] [answered]
项目中使用了大量内部 npm 包，均依赖 antd 4.24.x，如果项目升级到 antd 5，项目里的 5.x和npm包的4.x 共存的情况下时候会存在问题，有没有什么好的办法进行平滑过渡

**Comment by yoyo837** [comment]
没有 😄

**Comment by yoyo837** [comment]
建议先长期停留在4.x 满足业务, 找时间一次性升级上来.

**Comment by arvinxx** [comment, marked as answer]
真要搞的话，有个比较 hack 的方式

```json
{
  "antd":"^4.24.7",
  "antd-v5":"npm:antd@5"
}
```

然后 v5 组件从 `antd-v5` 里引入（并且建议用v5 的 ConfigProvider 改个prefix，否则样式会有冲突的）

等到组件库逐一从 v4 升到 v5 后，再把应用的 antd 版本升到v5。但中间各种问题就需要自行处理了，有好多坑的。

我也建议停留在4.x 满足业务, 然后找时间一次性升级上来
